### ISSUE TITLE
Parser events have offset, but not line number

### ISSUE THREAD

**Issue by zmughal** [perlpunk/YAML-PP-p5#52]
Is it possible to add the `$token->{line}` to parser events? I can extend the parser for now to work for most cases (will give an update if I notice anything missing).

My goal here is to have the loader maintain the location of YAML nodes alongside the constructed objects.

**Comment by perlpunk** [comment]
Yes, that's on my todo list. I just think that there are cases where I don't have that information (yet). I would need to have a closer look again, but busy for about a month at least...

**Comment by perlpunk** [comment]
Also I would like to be compatible to libyaml.
If you take a look at YAML::LIbYAML::API events, you will see detailed information about where an event starts and ends. So in the end I would ideally return the same information in YAML::PP

**Comment by NicolasRouquette** [comment]
> If you take a look at YAML::LIbYAML::API events, you will see detailed information about where an event starts and ends.

Is this the event information you're referring to?

https://github.com/yaml/libyaml/blob/master/include/yaml.h#L354-L478

The yaml 1.2.2 spec mentions "events" in the context of dump (section 3.1.1); unfortunately, the events are not explicitly defined.
